api/routes/Userdata.js

```diff
@@ -13,17 +12,11 @@ router.get("/all", async (req, res) => {
   res.status(200).json(userData);
 });
 
-router.get("/:id", async (req, res) => {
-  const { id } = req.params;
-
-  if (!mongoose.Types.ObjectId.isValid(id)) {
-    return res.status(404).json({ error: "No user found under this id." });
-  }
-
-  const userData = await User.findById(id);
-
+router.get("/:username", async (req, res) => {
+  const { username } = req.params;
+  const userData = await User.find({ name: username });
   if (!userData) {
```

Comment:
I don't recall what the `find` method returns, but if it returns an array you **can't** check it with `!userData` because you would be checking an empty array for its truthy value - and `Boolean([])` is always true. Conversely, `![]` is always false. You'd need to check for the length instead. This only applies for the case if `find` returns an array.